sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/WriteToDataSourceV2Exec.scala

```diff
@@ -501,3 +528,19 @@ private[v2] case class DataWritingSparkTaskResult(
  * Sink progress information collected after commit.
  */
 private[sql] case class StreamWriterCommitProgress(numOutputRows: Long)
+
+/**
+ * A trait that allows Tables that use V1 Writer interfaces to write data.
+ */
+sealed trait SupportsV1Write extends V2TableWriteExec {
+  def plan: LogicalPlan
+
+  protected def writeWithV1(
+      relation: CreatableRelationProvider,
+      mode: SaveMode,
+      options: CaseInsensitiveStringMap): RDD[InternalRow] = {
+    relation.createRelation(
+      sqlContext, mode, options.asScala.toMap, Dataset.ofRows(sqlContext.sparkSession, plan))
+    sparkContext.emptyRDD
```

Comment:
I might be missing something here, but why does this return an RDD and why is it empty?

Reply:
Ok so I see this is used in Atomic Table Writes, that seems a bit wrong to me, shouldn't we just not support the atomic table writes with V1 Fallback? Seems like we are violating the contract by returning empty regardless of what happens.

Reply:
What is the contract? All data write commands have returned an empty result since the beginning of time, e.g. look at `SaveIntoDataSourceCommand`, `V2TableWriteExec`, `InsertIntoDataSourceCommand`.

Reply:
```          val writtenRows = writer match {
            case v1: V1WriteBuilder =>
              writeWithV1(v1.buildForV1Write(), writeOptions)
            case v2 =>
              doWrite(v2.buildForBatch())
          }```
If this is always empty why do we save it as writtenRows here? This is just to hold a reference to the empty result set?

Reply:
yeah. It's pretty much dead code. I think if we decide to change what to return later, it's easier to change 1-2 places vs 'n' different operators.